pkg/natdiscovery/natdiscovery.go

```diff
@@ -0,0 +1,176 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package natdiscovery
+
+import (
+	"crypto/rand"
+	"math/big"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/types"
+)
+
+type Interface interface {
+	Run(stopCh <-chan struct{}, natResultCh chan RemoteEndpointNAT) error
+	AddEndpoint(endpoint *types.SubmarinerEndpoint)
+	RemoveEndpoint(endpointName string)
+	SetReadyChannel(readyChannel chan *RemoteEndpointNAT)
```

Comment:
As discussed on call, let's keep `RemoteEndpointNAT` private and expose a smaller struct, eg

```
type NATEndpointInfo struct {
    Endpoint  *types.SubmarinerEndpoint
    UseNAT   bool
    UseIP       string
}
```

Reply:
done.